pkg/server/node.go

```diff
@@ -188,14 +189,16 @@ func (s *nodeServer) fetchRegistrationEntries(selectors []*common.Selector, spif
 	var entries []*common.RegistrationEntry
 
 	///lookup Registration Entries for resolved selectors
-	req := &datastore.ListSelectorEntriesRequest{Selectors: selectors}
-	listSelectorResponse, err := dataStore.ListMatchingEntries(req)
-	if err != nil {
-		return nil, err
-	}
+	for combination := range selector.PowerSet(selector.NewSet(selectors)) {
```

Comment:
Tests to capture this updated logic would be awesome... I feel we'll probably refactor this handler in the near future

Reply:
Added test.